src/storage/src/source/sql_server/replication.rs

```diff
@@ -327,7 +327,12 @@ pub(crate) fn render<G: Scope<Timestamp = Lsn>>(
                         .expect("resume_upper has at least one value");
                     (Arc::clone(&src_info.capture_instance), resume_lsn)
                 })
-                .collect();
+                .fold(BTreeMap::new(), |mut map, (capture_instance, resume_lsn)| {
+                    map.entry(Arc::clone(&capture_instance))
+                    .and_modify(|existing| *existing = std::cmp::min(*existing, resume_lsn))
+                    .or_insert(resume_lsn);
+                    map
+                });
```

Comment:
PR looks great, but this is starting to look not-rusty imo. The imperative version is what I would expect to see in idiomatic Rust. Also added a little antichain goodness since we're in `TotalOrder` land.

```rust
let mut resume_lsns = BTreeMap::new();
for src_info in outputs.values() {
    let resume_lsn = match src_info.resume_upper.as_option() {
        Some(lsn) if *lsn != Lsn::minimum() => *lsn,
        Some(_) =>  src_info.initial_lsn.increment(),
        None => panic!("resume_upper has at least one value"),
    };
    resume_lsns.entry(Arc::clone(&src_info.capture_instance))
        .and_modify(|lsn| *lsn = std::cmp::min(*lsn, resume_lsn))
        .or_insert(resume_lsn);
}
```